temporalio/worker/_workflow_instance.py

```diff
@@ -1763,6 +1780,7 @@ def _run_once(self, *, check_conditions: bool) -> None:
                     ]
         finally:
             asyncio._set_running_loop(None)
+            self._current_thread_id = None
```

Comment:
This isn't a hig deal but it feels like we should be able to implement this in a way that keeps these two "in sync" automatically. I.e., we have an invariant: one of the following is always true:

- workflow thread ID is set and asyncio has a running loop
- workflow thread ID is not set and asyncio doesn't have a running loop

Can we maybe design the getter so that it always returns `None` if there's no running loop? Something alomg those lines.

Reply:
Thread is unrelated to running loop IMO, so I commented above that I think thread ID set/unset should be set upon activate not run_once (unless there is a reason I forgot to why we moved it here).